### PR TITLE
fix(cli/upload): GithubStatus check fix failedResults/warnResults

### PR DESCRIPTION
**Pull request by raphaelbs**
This issue happens when `includePassedAssertions` is set as `true`.

The first commit shows the error as the test was written before the fix.

Relates to issue #570
Inspired by #571

### CODE REVIEW

**Review by patrickhulce — CHANGES_REQUESTED** [review]
thank you so much for this great PR @raphaelbs!

apologies for the delay but I've just returned from leave, future reviews should be faster :)

File: packages/cli/src/upload/upload.js [hunk 1]
```diff
@@ -165,6 +165,9 @@ async function postStatusToGitHub(options) {
   } else {
     const url = `${githubApiHost}/repos/${slug}/statuses/${hash}`;
     const payload = {state, context, description, target_url: targetUrl};
+    if (process.env.NODE_ENV === 'test') {
+      console.debug(payload);
+    }
```
[the remark this answers is not in the document]
Comment (raphaelbs) [reply packages/cli/src/upload/upload.js:170]: The debug feels like a dirty hack haha
I'll look into the server fixture

**Comment by patrickhulce** [comment]
@raphaelbs how are we looking here? need any help?

**Comment by raphaelbs** [comment]
> @raphaelbs how are we looking here? need any help?

Sorry, I couldn't look it into yet.

**Comment by metalix2** [comment]
> > @raphaelbs how are we looking here? need any help?
> 
> Sorry, I couldn't look it into yet.

More than happy to jump in and propose a PR this with the additional changes.. I've been watching this closely the last month. 😅

**Comment by patrickhulce** [comment]
How does that sound @raphaelbs! If you don't have the bandwidth to finish it up this week, maybe @metalix2 could lend a hand?

Greatly appreciate the work so far, and it's almost there just want to help get it across the finish line! 🏁

**Comment by raphaelbs** [comment]
> More than happy to jump in and propose a PR this with the additional changes.. I've been watching this closely the last month.

That would be great! I appreciate it @metalix2

**Comment by metalix2** [comment]
> > More than happy to jump in and propose a PR this with the additional changes.. I've been watching this closely the last month.
> 
> That would be great! I appreciate it @metalix2

Excellent PR raised! 👍

**Comment by raphaelbs** [comment]
finishing on #595

**Pull request closed without merging**